src/common/expandVariants.ts

```diff
@@ -1,108 +1,42 @@
 import { IR } from "../IR";
-import { Path, programToPath } from "./traverse";
+import {
+  fromChildRemapFunc,
+  getChild,
+  getChildFragments,
+  PathFragment,
+} from "./fragments";
 
 /**
  * Expand all of the variant nodes in program to get a list of fully-
  * instantiated Programs (without any Variant nodes in them)
  */
 export function expandVariants(program: IR.Program): IR.Program[] {
-  const structure: Variant = getVariantsStructure(program);
-  const expansionCount = countVariantExpansions(structure);
-  if (expansionCount > 1000) {
-    throw new Error(`Too many variants (${expansionCount}).`);
+  return allVariantOptions(program) as IR.Program[];
+}
+
+function allVariantOptions(node: IR.Node): IR.Node[] {
```

Comment:
Previously, the total amount of expanded variants was calculated before the actual expansion happened so that the expansion is not performed when it was too high. Creating more variants than what can be processed in a reasonable time (or even fit in memory) probably cannot happen by accident, but still, I think it was a nice feature to have.